lib/pos_mempool.go

```diff
@@ -834,9 +825,6 @@ func (mp *PosMempool) UpdateGlobalParams(globalParams *GlobalParamsEntry) {
 	}
 
 	mp.globalParams = globalParams
-	if err := mp.refreshNoLock(); err != nil {
```

Comment:
I think we need a refresh here. @tholonious - I think we were relying on a call to `UpdateGlobalParams` in the `OnBlockConnected` event to remove txns that were in a block from the mempool.

Reply:
I've noticed there's also a separate method in PosMempool called `OnBlockConnected` which is doing something similar to what you're describing. Maybe that method is sufficient?

Also, interestingly, I think Refresh will eventually remove the stale transactions anyway since they'll fail to connect if they were already included in the chain.